cli/src/main/java/com/containersol/minimesos/main/Main.java

```diff
@@ -67,69 +69,87 @@ public static void main(String[] args) {
         }
     }
 
-    public Main() {
-        jc = new JCommander(this);
-        jc.setProgramName("minimesos");
-    }
 
     public void setOutput(PrintStream output) {
         this.output = output;
     }
 
     public int run(String[] args) {
+        initJCommander();
+
+        try {
+            parseParams(jc, args);
 
+            if (help) {
+                printUsage(null);
+                return EXIT_CODE_OK;
+            }
+
+            if (debug) {
+                initializeDebugLogging();
+            }
+
+            if (jc.getParsedCommand() == null) {
+                return handleNoCommand();
+            }
+
+            if (!commands.containsKey(jc.getParsedCommand())) {
+                LOGGER.error("Unknown command: " + jc.getParsedCommand());
+                return EXIT_CODE_ERR;
+            }
+
+            Command parsedCommand = commands.get(jc.getParsedCommand());
+            if (CommandHelp.CLINAME.equals(parsedCommand.getName())) {
+                printUsage(null);
+            } else {
+                if (parsedCommand.validateParameters()) {
+                    parsedCommand.execute();
+                } else {
+                    printUsage(jc.getParsedCommand());
+                    return EXIT_CODE_ERR;
+                }
+            }
+
+            return EXIT_CODE_OK;
+        } catch (Exception ex) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3AS1166)